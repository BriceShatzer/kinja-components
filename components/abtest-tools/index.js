/* @flow */

import ABTestIframes from './abtest-iframes';

export default ABTestIframes;
