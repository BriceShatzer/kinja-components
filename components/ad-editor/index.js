// @flow

import * as React from 'react';
import { useState, useEffect } from 'react';

import styled from 'styled-components';

import media from 'kinja-components/style-utils/media';

const PANELS = [
	{
		name: 'Primary Panel',
		resources: [
			{
				name: '16:9 Aspect Ratio',
				resource: 'https://app.specless.tech/api/1/creative/snapshot/ptc8tD/0006/primary-panel.png?q=low'
			},
			{
				name: '1:1 Aspect Ratio',
				resource: 'https://app.specless.tech/api/1/creative/snapshot/ptc8tD/0006/primary-panel/1by1.png?w=450&h=450&q=low'
			},
			{
				name: '3:1 Aspect Ratio',
				resource: 'https://app.specless.tech/api/1/creative/snapshot/ptc8tD/0006/primary-panel/3by1.png?w=1200&h=400&q=low'
			}
		]
	},
	{
		name: 'Leave Behind',
		resources: [
			{
				name: 'Large',
				resource: 'https://serve.specless.tech/upload/1/creative/_opt:' +
						'c_crop,f_png,w_2793,h_152,x_988,y_2642/ptc8tD/assets/' +
						'ck7bu0f82000f___egcj8r9a_NationalGeographic_Superhero.png'
			},
			{
				name: 'Small',
				resource: 'https://serve.specless.tech/upload/1/creative/_opt:' +
						'c_crop,f_png,w_1903,h_152,x_1433,y_3035/ptc8tD/assets/' +
						'ck7bu0f82000f___egcj8r9a_NationalGeographic_Superhero.png'
			},
			{
				name: 'Wide',
				resource: 'https://serve.specless.tech/upload/1/creative/_opt:' +
						'c_crop,f_png,w_4187,h_151,x_291,y_2250/ptc8tD/assets/' +
						'ck7bu0f82000f___egcj8r9a_NationalGeographic_Superhero.png'
			}
		]
	}
];

const UnstyledHeader = ({ className, children }) => {
	return (
		<header className={className}>
			{children}
			<hr />
		</header>
	);
};
const Header = styled(UnstyledHeader)`
	width: 100%;

	h5 {
		text-align: center;
		margin: 0.5rem;
	}
	hr {
		margin-top: 0;
	}
`;

const UnstyledAdEditorSection = ({className, title, children }) => {
	return (<article className={ className }>
		<Header><h5>{ title }</h5></Header>
		{ children }
	</article>);
};
const AdEditorSection = styled(UnstyledAdEditorSection)`
	box-sizing: border-box;

	flex-grow: ${({ flexGrow }) => flexGrow};
	height: auto;
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
	align-items: center;

	box-shadow: 2px 0 5px 0 rgb(110, 110, 110);
`;

const UnstyledElem = ({className, title, imgUrl, onClick }) => {
	return (
		<section className={className} onClick={onClick}>
			<div>
				<Background />
				<picture>
					<img src={imgUrl} />
				</picture>
			</div>

			<header><h6>{ title }</h6></header>
		</section>
	);
};
const Elem = styled(UnstyledElem)`
	width: 112px;
	height: 112px;

	${media.largeUp`
		width: 130px;
    	height: 130px;
	`}

	${media.xlargeUp`
		width: 160px;
    	height: 160px;
	`}

	display: flex;
	flex-flow: column;
	justify-content: flex-start;
	align-items: center;

	> div {
		position: relative;
		width: 70%;
		height: 70%;
		display: flex;
		justify-content: center;
		flex-direction: column;
	}

	> header {
		border: 0;
		margin: 0;
		padding: 0.4rem;
		text-align: center;
		height: 30%;
		width: 100%;
	}

	> header h6 {
		font-size: 0.8rem;
		font-weight: thin;
		margin: 0;
	}
`;

const Background = styled.div`
	position: absolute;
	z-index: -1;
	width: 100%;
	height: 100%;
	border: 0;

	background-image: url('https://app.specless.tech/app/assets/img/checker.png');
	background-repeat: repeat;
	opacity: 0.05;
`;

const UnstyledPreview = ({ className, name, resource }) => (
	<section className={className}>
		<header><h6>{name}</h6></header>
		<picture>
			<img src={resource}/>
		</picture>
	</section>
);

const Preview = styled(UnstyledPreview)`
	h6 {
		padding-left: 1rem;
	}
`;

export type PANEL = {
    name: string;
    resources: {
        name: string;
        resource: string;
    }[];
}

export const AdEditor = ({ panels = PANELS }: { panels?: PANEL[] }) => {
	const [selectedPanel, setSelectedPanel] = useState(null);
	const [selectedLayout, setSelectedLayout] = useState(null);

	useEffect(setDefaultPanel, []);
	useEffect(setSelectedPanelDefaultLayout, [panels, selectedPanel]);

	return (
		<>
			<AdEditorSection flexGrow={1} title="Panels">
				{
					panels.map(({ name, resources }) => (
						<Elem
							key={name}
							onClick={() => setSelectedPanel(name)}
							title={name}
							imgUrl={ resources.length ? resources[0].resource : '' }
						/>
					))
				}
			</AdEditorSection>
			<AdEditorSection flexGrow={1} title="Layouts">
				<div>
					{
						panels
							.filter(filterSelectedPanel)
							.reduce(flattenLayouts, [])
							.map(({ name, resource }) => (
								<Elem
									key={name}
									onClick={() => setSelectedLayout(name)}
									title={name}
									imgUrl={ resource }
								/>
							))
					}
				</div>
			</AdEditorSection>
			<AdEditorSection flexGrow={20} title="Preview">
				{
					panels
						.filter(filterSelectedPanel)
						.reduce(flattenLayouts, [])
						.filter(filterSelectedLayout)
						.map(({ name, resource }) => (
							<Preview key={name} name={name} resource={resource}/>
						))
				}
			</AdEditorSection>
		</>
	);

	function setSelectedPanelDefaultLayout() {
		const layouts = panels.filter(filterSelectedPanel).reduce(flattenLayouts, []);
		const defaultLayout = layouts.length > 0 ? layouts[0] : null;

		if (!defaultLayout) { return; }
		setSelectedLayout(defaultLayout.name);
	}

	function setDefaultPanel() {
		const defaultPanel = panels.length > 0 ? panels[0] : null;

		if (!defaultPanel) { return; }
		setSelectedPanel(defaultPanel.name);
	}

	function filterSelectedPanel({ name }) {
		return name === selectedPanel;
	}

	function flattenLayouts(accum, { resources }) {
		return [...accum, ...resources];
	}

	function filterSelectedLayout({ name }) {
		return name === selectedLayout;
	}
};
