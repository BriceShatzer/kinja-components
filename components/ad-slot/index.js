/* @flow */

import AdSlot from './ad-slot';

export default AdSlot;
