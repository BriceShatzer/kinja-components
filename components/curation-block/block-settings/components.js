// @flow
import styled from 'styled-components';
import media from '../../../style-utils/media';
import { gridValue } from '../../grid-utils';
import { Wrapper as RadioWrapper } from '../../form/radio';

export const SectionHeader = styled.h4`
	color: ${props => props.theme.color.darkgray};
	font-size: ${props => props.theme.typography.primary.fontSizes.small};
	line-height: ${props => props.theme.typography.primary.lineHeights.small};
	margin-bottom: 1.5rem;
`;

export const FieldHeaderLabel = styled.label`
	color: ${props => props.theme.color.gray};
	font-size: ${props => props.theme.typography.primary.fontSizes.xsmall};
	line-height: ${props => props.theme.typography.primary.lineHeights.xsmall};
	margin-bottom: 1.5rem;
`;

export const Container = styled.div`
	margin-bottom: 2rem;
`;

export const ColumnWrapper = styled.div`
	${media.xlargeUp`
		display: flex;
	`}
`;

export const Column = styled.div`
	width: ${gridValue.small('6c')};
	margin-bottom: 1rem;
	${media.xlargeUp`
		margin-bottom: 0;
	`}

	${media.xlargeOnly`
		width: ${gridValue.xlarge('3c')};
		&:not(:last-child) {
			margin-right: ${gridValue.xlarge('1g')};
		}
	`}

	${media.xxlargeOnly`
		width: ${gridValue.xxlarge('3c')};
		&:not(:last-child) {
			margin-right: ${gridValue.xxlarge('1g')};
		}
	`}

	${media.xxxlargeUp`
		width: ${gridValue.xxxlarge('3c')};
		&:not(:last-child) {
			margin-right: ${gridValue.xxxlarge('1g')};
		}
	`}

	${RadioWrapper} {
		height: auto;
	}
`;