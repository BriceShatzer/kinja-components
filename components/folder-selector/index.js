/* @flow */

import FolderSelector, {
	BrowserContainer,
	BrowserWrapper,
	Wrapper
} from './folder-selector';

export default FolderSelector;

// needed by require.ensure
export {
	FolderSelector,
	BrowserContainer,
	BrowserWrapper,
	Wrapper
};
