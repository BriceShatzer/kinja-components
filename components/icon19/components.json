{
  "18x18": [
    "AddBlog",
    "AlignBottomCenter",
    "AlignBottomLeft",
    "AlignBottomRight",
    "AlignCenter",
    "AlignLeft",
    "AlignMiddleCenter",
    "AlignRight",
    "ArrowBottomLeft",
    "ArrowBottomRight",
    "ArrowBottom",
    "ArrowLeft",
    "ArrowRight",
    "ArrowTopLeft",
    "ArrowTopRight",
    "ArrowTop",
    "Bell",
    "Bleed",
    "Blog",
    "Bold",
    "Bookmark",
    "BookmarkFilled",
    "Breaking",
    "Bubble",
    "BubbleAdd",
    "BubbleDelete",
    "BubbleFilled",
    "BubbleRemove",
    "Burner",
    "Cart",
    "CartButton",
    "Center",
    "Checkmark",
    "ChevronDown",
    "ChevronLeft",
    "ChevronRight",
    "ChevronUp",
    "Clock",
    "Close",
    "Code",
    "Collapse",
    "ColorPicker",
    "Compare",
    "Copy",
    "Copyright",
    "Crop",
    "CurationTertiaryFour",
    "CurationTertiaryOne",
    "Disabled",
    "DollarCircle",
    "Dollar",
    "DoubleChevronDown",
    "DoubleChevronLeft",
    "DoubleChevronRight",
    "DoubleChevronUp",
    "DropCap",
    "Editor",
    "Ellipsis",
    "EmbedThumb",
    "Embiggen",
    "EmbiggenFilled",
    "ErrorFilled",
    "Error",
    "Expand",
    "ExpandBackground",
    "ExternalLink",
    "ExternalLinkError",
    "Eye",
    "EyeFilled",
    "Facebook",
    "FeaturedHeaderClassic",
    "FeaturedHeaderClassicReversed",
    "FeaturedHeaderImpact",
    "FeaturedHeaderMagazine",
    "Flag",
    "Flame",
    "Flask",
    "Folder",
    "FolderVideo",
    "FullBleedWidget",
    "Fullscreen",
    "Gear",
    "Globe",
    "Google",
    "GoogleColor",
    "Hamburger",
    "Hashmark",
    "Heart",
    "Home",
    "HorizontalRule",
    "Image",
    "Inbox",
    "Information",
    "InsertModule",
    "InsetMinimal",
    "Instagram",
    "Issue",
    "Italic",
    "Key",
    "Kinja",
    "Label",
    "LayoutHorizontal",
    "LayoutVertical",
    "Left",
    "Lightning",
    "Link",
    "Linkedin",
    "Loading",
    "Lock",
    "LockFilled",
    "Login",
    "Logout",
    "Mail",
    "MailFilled",
    "Minus",
    "MirrorHorizontal",
    "MirrorVertical",
    "MomJeans",
    "MoveBottom",
    "MoveLeft",
    "MoveRight",
    "MoveTop",
    "MoveVertical",
    "OrderNewest",
    "OrderOldest",
    "OrderPopular",
    "OrderedList",
    "OverlayDark",
    "OverlayLight",
    "Pagebreak",
    "Parallax",
    "Pause",
    "PauseFilled",
    "Pencil",
    "Pin",
    "Play",
    "PlayFilled",
    "Plus",
    "Promotion",
    "Quotable",
    "Quote",
    "RSS",
    "RecommendedStories",
    "Reload",
    "ReplyLeft",
    "ReplyRight",
    "Review",
    "Right",
    "Rotate",
    "Search",
    "SearchVideo",
    "Settings",
    "Share",
    "SharedPost",
    "Slideshow",
    "SortAlphabetically",
    "Star",
    "StarCircle",
    "StarFilled",
    "Strikethrough",
    "TextColorDark",
    "TextColorLight",
    "TileBackground",
    "Trashcan",
    "Twitter",
    "Redo",
    "Undo",
    "UnorderedList",
    "User",
    "UserAdd",
    "UserAvatar",
    "UserFilled",
    "UserRemove",
    "Users",
    "Video",
    "VolumeMax",
    "Write",
    "Youtube",
    "ZoomOut",
    "ZoomIn"
  ],
  "24x24": [
    "Blog24",
    "Bookmark24",
    "BookmarkFilled24",
    "Bubble24",
    "Clock24",
    "Close24",
    "DealsBox24",
    "Flame24",
    "Hamburger24",
    "Label24",
    "Promotion24",
    "Search24",
    "SharedPost24",
    "Tools24",
    "Video24"
  ],
  "48x48": [
    "CurationEqualFour",
    "CurationEqualSix",
    "CurationEqualThree",
    "CurationHeadline",
    "CurationModularFour",
    "CurationModularThree",
    "Curation20Ad",
    "Curation20EightCard",
    "Curation20FeaturedStory",
    "Curation20FiveCard",
    "Curation20FiveCardTwoColumn",
    "Curation20FourCard",
    "Curation20FourCardHorizontal",
    "Curation20FourCardHorizontalCompact",
    "Curation20SingleStory",
    "Curation20SixCardGrid",
    "Curation20SixCardWithLatest",
    "Curation20ThreeCardHorizontal",
    "Curation20Video",
    "DealsBeauty",
    "DealsGaming",
    "DealsHome",
    "DealsLifestyle",
    "DealsMultimedia",
    "DealsTech",
    "DealsTop",
    "LunchboxHeadline",
    "LunchboxHeadlineHero",
    "LunchboxHeadlineImage",
    "LunchboxHeadlineText",
    "LunchboxHeadlineTextHero",
    "LunchboxHeadlineTextImage",
    "LunchboxHeroHeadline",
    "LunchboxHeroHeadlineText",
    "LunchboxHeroText",
    "LunchboxImage",
    "LunchboxImageFullbleed",
    "LunchboxImageHeadline",
    "LunchboxImageHeadlineText",
    "LunchboxImageParallax",
    "LunchboxImageText",
    "LunchboxText",
    "LunchboxTextHero",
    "LunchboxTextImage"
  ]
}