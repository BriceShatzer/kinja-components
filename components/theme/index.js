// @flow

import Theme from './theme';
import { matchBlogGroupToTheme } from './theme';
import defaultTheme from './themes';
import EnsureDefaultTheme from './ensureDefaultTheme';

export {
	Theme,
	defaultTheme,
	EnsureDefaultTheme,
	matchBlogGroupToTheme
};
export default Theme;
