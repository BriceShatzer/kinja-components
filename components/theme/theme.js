// @flow

import * as React from 'react';
import { ThemeProvider } from 'styled-components';

import { blogThemes as Themes } from './themes';

export type BlogThemeName = $Keys<typeof Themes>;
const blogThemeNames: Array<BlogThemeName> = Object.keys(Themes);

type Props = {
	children: React.Node,
	theme: {},
	blog?: ?BlogThemeName
};

export const matchBlogGroupToTheme = (blogGroup: ?string): BlogThemeName => {
	return (blogThemeNames.find(b => b === blogGroup)) || 'default';
};

const Theme = (props: Props) => {
	const { children, theme, blog } = props;

	const appliedTheme = blog ? {
		...(Themes[blog] || Themes.default),
		blog
	} : theme;

	return (
		<ThemeProvider theme={appliedTheme}>
			{children}
		</ThemeProvider>
	);
};

Theme.defaultProps = {
	theme: Themes.default
};

export default Theme;
