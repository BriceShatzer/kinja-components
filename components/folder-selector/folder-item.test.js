/* @flow */

import * as React from 'react';
import { shallow, mount } from 'enzyme';
import uuid from 'uuid/v4';
import MomJeansIcon from '../icon19/MomJeans';
import { EnsureDefaultTheme } from 'kinja-components/components/theme';
import FolderItem, {Wrapper} from './folder-item';

const onSelect = jest.fn();
const id = uuid();

describe('<FolderItem/>', () => {
	it('should render with default icon', () => {
		const folderItem = shallow(<FolderItem name="Barf Bag" id={id} onSelect={() => {}} />);

		expect(folderItem).toMatchSnapshot();
	});

	it('should render with default icon', () => {
		const folderItem = shallow(<FolderItem name="Barf Bag" id={id} onSelect={onSelect} />);

		expect(folderItem).toMatchSnapshot();
	});

	it('should render with default icon for video', () => {
		const folderItem = shallow(<FolderItem name="Barf Bag" id={id} onSelect={onSelect} isVideo />);

		expect(folderItem).toMatchSnapshot();
	});

	it('should render chevron for items with children', () => {
		const folderItem = shallow(<FolderItem name="Barf Bag" id={id} onSelect={onSelect} hasChildren />);

		expect(folderItem).toMatchSnapshot();
	});

	it('should render with icon passed in', () => {
		const folderItem = shallow(<FolderItem name="Barf Bag" id={id} onSelect={onSelect} icon={<MomJeansIcon />} />);

		expect(folderItem).toMatchSnapshot();
	});

	it('should call onSelect with id', () => {
		const folderItem = shallow(<FolderItem name="Barf Bag" id={id} onSelect={onSelect} />);
		folderItem.find(Wrapper).simulate('click');

		expect(onSelect).toBeCalledWith(id);
	});

	it('should render with checkmark prefix', () => {
		const folderItem = mount(
			<EnsureDefaultTheme>
				<FolderItem isFirst name="Barf Bag" id={id} onSelect={onSelect} multipleSelection selected />
			</EnsureDefaultTheme>);

		expect(folderItem.find('CheckmarkIcon').exists()).toBe(true);
	});

	it('should render without prefix', () => {
		const folderItem = mount(
			<EnsureDefaultTheme>
				<FolderItem isFirst name="Barf Bag" id={id} onSelect={onSelect} multipleSelection />
			</EnsureDefaultTheme>);
		expect(folderItem.find('CheckmarkIcon').exists()).toBe(false);
	});
});
