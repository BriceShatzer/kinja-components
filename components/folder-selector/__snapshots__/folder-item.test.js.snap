// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FolderItem/> should render chevron for items with children 1`] = `
<folder-item__Wrapper
  data-shouldshowtooltip={false}
  data-tooltipname="Barf Bag"
  data-tooltipoffset={5}
  onClick={[Function]}
>
  <folder-item__Inner>
    <folder-item__IconWrapper>
      <FolderIcon />
    </folder-item__IconWrapper>
    <folder-item__Name>
      Barf Bag
    </folder-item__Name>
    <ChevronRightIcon />
  </folder-item__Inner>
</folder-item__Wrapper>
`;

exports[`<FolderItem/> should render with default icon 1`] = `
<folder-item__Wrapper
  data-shouldshowtooltip={false}
  data-tooltipname="Barf Bag"
  data-tooltipoffset={5}
  onClick={[Function]}
>
  <folder-item__Inner>
    <folder-item__IconWrapper>
      <FolderIcon />
    </folder-item__IconWrapper>
    <folder-item__Name>
      Barf Bag
    </folder-item__Name>
  </folder-item__Inner>
</folder-item__Wrapper>
`;

exports[`<FolderItem/> should render with default icon 2`] = `
<folder-item__Wrapper
  data-shouldshowtooltip={false}
  data-tooltipname="Barf Bag"
  data-tooltipoffset={5}
  onClick={[Function]}
>
  <folder-item__Inner>
    <folder-item__IconWrapper>
      <FolderIcon />
    </folder-item__IconWrapper>
    <folder-item__Name>
      Barf Bag
    </folder-item__Name>
  </folder-item__Inner>
</folder-item__Wrapper>
`;

exports[`<FolderItem/> should render with default icon for video 1`] = `
<folder-item__Wrapper
  data-shouldshowtooltip={false}
  data-tooltipname="Barf Bag"
  data-tooltipoffset={5}
  onClick={[Function]}
>
  <folder-item__Inner>
    <folder-item__IconWrapper>
      <FolderVideoIcon />
    </folder-item__IconWrapper>
    <folder-item__Name>
      Barf Bag
    </folder-item__Name>
  </folder-item__Inner>
</folder-item__Wrapper>
`;

exports[`<FolderItem/> should render with icon passed in 1`] = `
<folder-item__Wrapper
  data-shouldshowtooltip={false}
  data-tooltipname="Barf Bag"
  data-tooltipoffset={5}
  onClick={[Function]}
>
  <folder-item__Inner>
    <folder-item__IconWrapper>
      <MomJeansIcon />
    </folder-item__IconWrapper>
    <folder-item__Name>
      Barf Bag
    </folder-item__Name>
  </folder-item__Inner>
</folder-item__Wrapper>
`;
